test/blackbox-tests/test-cases/pkg/odoc/dev-tool-odoc-no-ocaml-lockfile.t

```diff
@@ -14,7 +14,6 @@ doesn't contain a lockfile for the "ocaml" package.
   $ make_lockdir
 
   $ DUNE_CONFIG__LOCK_DEV_TOOL=enabled dune ocaml doc
-  Error: The lockdir doesn't contain a lockfile for the package "ocaml".
-  Hint: Add a dependency on "ocaml" to one of the packages in dune-project and
-  then run 'dune pkg lock'
+  Error: No compiler declared in the lockfile
+  Hint: Try running 'dune build'
```

Comment:
How does running `dune build` help in this situation?

Reply:
Why can't we just pick any compiler at this point? Surely if there is no compiler in the project, then it doesn't matter what we pick?

Reply:
If the project has no compiler, then there is no point in locking a dev-tool that needs a matching compiler. It won't work and once the project gets a compiler we need to re-lock the dev-tool and use that compiler anyway.